wordpress-mobile/WordPress-iOS
# Edit Comment: save changes

Ref: #17000 
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/440

When `Done` is selected, the Comment changes are now saved.
- A `completion` closure is called when `Done` or `Cancel` is tapped. It returns the Comment object and a Bool indicating if the Comment has been changed.
- The WPKit PR has changes to include the author's name, email, and url when updating a Comment.

To test:

---
- Enable the `newCommentEdit` feature.
- Go to My Site > Comments > Comment details > Edit.
- Modify all the fields.
- Select `Done`.
  - Verify the author name and web address are updated in the comment header.
  - Verify the comment content is updated in the comment details.
- Select `Edit` again. 
  - Verify all the info is the updated version.
- Repeat for a self-hosted site.

| Initial | Edited | Updated |
|--------|-------|-------|
| ![before_details](https://user-images.githubusercontent.com/1816888/131393126-e02a08d8-85fc-46d6-893c-c9b0992c5f1f.png) | ![edited](https://user-images.githubusercontent.com/1816888/131393137-330f85c1-0534-4a42-80d3-362d6135c8ba.png) | ![after_details](https://user-images.githubusercontent.com/1816888/131393158-d19b16a7-a9de-49ab-b73b-4a6e759bc551.png) |

---
- Disable `newCommentEdit`.
- Edit a Comment.
- Verify it is updated as expected.

---
## Regression Notes
1. Potential unintended areas of impact
N/A. Feature is incomplete and disabled.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A. Feature is incomplete and disabled.

3. What automated tests I added (or what prevented me from doing so)
N/A. Feature is incomplete and disabled.

PR submission checklist:
